samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/samples/FileTransferIntegrationTest.java

```diff
@@ -0,0 +1,253 @@
+                                .isEqualTo(TransferProcessStates.COMPLETED.code())
+                ));
+
+        var copiedFilePath = Path.of(format(CONSUMER_ASSET_PATH + "/%s.txt", PROVIDER_ASSET_NAME));
+        var actualFileContent = fetchFileContent(copiedFilePath);
+        assertThat(actualFileContent)
+                .withFailMessage("Transferred file contents are null")
+                .isNotNull();
+        assertThat(actualFileContent)
+                .withFailMessage("Transferred file contents are not same as the source file")
+                .isEqualTo(fileContent);
+    }
+
+    private ObjectNode fetchTransfer(String transferProcessId) {
+        return
+                given()
+                        .pathParam(TRANSFER_ID_PARAM, transferProcessId)
+                .when()
+                        .get(TRANSFER_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().as(ObjectNode.class);
+    }
+
+    /**
+     * Fetch negotiated contract agreement.
+     *
+     * @param contractNegotiationRequestId ID of the ongoing contract negotiation between consumer and provider.
+     * @return Negotiation as {@link ObjectNode}.
+     */
+    private ObjectNode fetchNegotiatedAgreement(String contractNegotiationRequestId) {
+        return
+                given()
+                        .pathParam(CONTRACT_NEGOTIATION_REQUEST_ID_PARAM, contractNegotiationRequestId)
+                        .header(API_KEY_HEADER, API_KEY_CONTROL_AUTH)
+                .when()
+                        .get(CONTRACT_AGREEMENT_PATH)
+                .then()
+                        .assertThat().statusCode(HttpStatus.SC_OK)
+                        .extract().as(ObjectNode.class);
+    }
+
+    /**
+     * Helper method to read file contents on the given {@link Path}
+     *
+     * @param filePath see {@link Path}
+     * @return Contents of file as a {@link String} or null if file does not exist.
+     */
+    private String fetchFileContent(Path filePath) {
+        if (filePath.toFile().exists()) {
+            try {
+                return Files.readAllLines(filePath).get(0);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Helper method to create a temporary directory.
+     *
+     * @return a newly create temporary directory.
+     */
+    private static String tempDirectory() {
+        try {
+            return Files.createTempDirectory(FileTransferIntegrationTest.class.getSimpleName()).toString();
+        } catch (IOException e) {
+            throw new EdcException(e);
+        }
+    }
+
+    /**
+     * Utility method to find an unallocated port. Note that there is a race condition,
+     * the port might be allocated by the time it is used.
+     *
+     * @return a server port.
+     */
+    private static int findUnallocatedServerPort() {
```

Comment:
But do you see it belongs more to EdcRuntimeExtension ? It was kept here as it is for the test and not for the Junit extension.

Thoughts ?

Reply:
IMO this is something that all tests will need, correct? SpringBoot has the [RANDOM_PORT](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/test/context/SpringBootTest.WebEnvironment.html#RANDOM_PORT) together with the [@LocalServerPort](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/web/server/LocalServerPort.html) annotation for that. We don't need something that fancy, but it would be good to have it centralized in the testing framework.

Reply:
That method actually already exists elsewhere in EDC tests, could be factored out to a utility class

Reply:
Moved to common/util/src/main/java/org/eclipse/dataspaceconnector/common/configuration/ConfigurationFunctions.java with a simple test.